roles/upload-site/tasks/main.yml

```diff
@@ -41,32 +41,52 @@
 # per-site basis. That will allow us to move past errors in this task, and run
 # "drush updb" a second time in the following task.
 - name: Update database schema with drush updb
-  shell: "{{drush_alias }} {{ item }}"
+  shell: "{{ drush_alias }} {{ item }}"
   with_items:
     - "-y updb"
   ignore_errors: "{% if ignore_updb_errors == 'TRUE' %}yes{% endif %}"
 
+# This only runs "drush rr" if ignore_updb_errors == "TRUE"
+- name: Run drush rr before running drush updb a second time for recalcitrant sites
+  shell: "{{ drush_alias }} {{ item }}"
+  with_items:
+    - "-y rr"
+  when: ignore_updb_errors == "TRUE"
+  ignore_errors: yes
+
 # This only runs "drush updb" if ignore_updb_errors == "TRUE"
 - name: Run drush updb a second time for recalcitrant sites
-  shell: "{{drush_alias }} {{ item }}"
+  shell: "{{ drush_alias }} {{ item }}"
   with_items:
     - "-y updb"
   when: ignore_updb_errors == "TRUE"
 
+# Set the install profile by looking for the enabled supporting module.
+- name: Find and set the install profile by helper modules.
```

Comment:
New function for products. Used swsbuildjsa as an example site to migrate.

Reply:
https://swsbuildjsa-dev.cardinalsites.stanford.edu/

Reply:
Can we assume a 1:1 here? Or shall we just assume a 1:1 gets us "close enough"?

Reply:
1:1 is pretty accurate. If the module is available and enabled (which is what module_exists checks) then we can be almost certain that this site is what it is we think it is.

Reply:
🤔 hmm.. I didn't think that we had anything that would be JSA without the helper module. That would throw a wrench into this mix.

Reply:
Initial commit May 1, 2014 for JSA

Reply:
Initial commit Dec 13, 2013 for JSA profile.

Reply:
Well.... shoot.

Reply:
Yeah, jsa-content would stay at the stanford profile... which is right imho as it really isn't a JSA.

Reply:
English would be set to stanford_jumpstart however...

Reply:
Ok, anything deployed after May 1st, 2014 can get the products group. Anything before... well... The likely need special treatment anyhow.

Reply:
Yeah that sounds about right. English was our build site for JSA, so by definition it wouldn't have `stanford_jumpstart_academic`. Art would be the other one; it only has `stanford_jumpstart` enabled but I *think* it's a JSA? 🤔

Reply:
1:1 gets us "close enough". We know that we have some ❄️ ❄️ ❄️

Reply:
The other approach to this that I was thinking of was to set it as a variable on a per-site basis (or put the sites into groups with an `install_profile` variable), rather than auto-detect. But since you've already done the work for auto-detect, and my approach would be more work. And I don't like work.

Reply:
Yeah, I think we can throw some spaghetti on the wall with this function and see what sticks. For the stuff that drops be a bit more explicit and run it with a set variable.